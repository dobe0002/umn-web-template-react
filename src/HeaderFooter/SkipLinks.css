.umnHeaderSkipLinksWrapperDiv {
  position: absolute;
  top: 0;
  left: -10000px;
  width: 1px;
  height: 1px;
  overflow: hidden;
  padding: 15px;
  background: #fff;
  border-radius: 0 0 5px 5px;
  box-shadow: 1px 2px 4px rgba(0, 0, 0, 0.5);
}

.umnHeaderSkipLinksWrapperDiv:focus-within {
  left: 0;
  width: auto;
  height: auto;
  overflow: auto;
}

.umnHeaderSkipLinksUl {
  margin: 0;
  padding: 0;
}

.umnHeaderSkipLinksLi {
  display: inline-block;
  margin-right: 15px;
}
