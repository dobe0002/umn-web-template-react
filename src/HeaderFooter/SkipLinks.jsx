import React from 'react';
// import styled from 'styled-components'; /* Not using styled-components so  styled-components isn't needed as a dependency */
// import PropTypes from 'prop-types';  /* Disabling this so prop-types isn't needed as a dependency */

import './SkipLinks.css'; /* Need to make a separate CSS  because it uses focus pseudo class */

const SkipLinks = (props) => {
  const { mainNavId, mainContentId } = props;
  if (props.mainNavId || props.mainContentId) {
    return (
      <div className='headerLinks umnHeaderSkipLinksWrapperDiv'>
        <ul className='umnHeaderSkipLinksUl'>
          {mainNavId && (
            <li className='umnHeaderSkipLinksLi'>
              <a href={`#${mainNavId}`}>Main navigation</a>
            </li>
          )}
          {mainContentId && (
            <li className='umnHeaderSkipLinksLi'>
              <a href={`#${mainContentId}`}>Main content</a>
            </li>
          )}
        </ul>
      </div>
    );
  }
  return '';
};

// SkipLinks.propTypes = {
//   mainNavId: PropTypes.string,
//   mainContentId: PropTypes.string,
// };

export default SkipLinks;
