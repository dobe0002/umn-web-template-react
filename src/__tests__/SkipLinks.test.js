/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import SkipLinks from '../HeaderFooter/SkipLinks';

const { axe, toHaveNoViolations } = require('jest-axe');

expect.extend(toHaveNoViolations);
configure({ adapter: new Adapter() });

describe('Skip Links', () => {
  it('Accessibility check', async () => {
    const temp = (
      <header>
        <SkipLinks mainNavId='foo' mainContentId='bar' />
        <a id='foo' href='item1'>
          item1
        </a>
        <a id='bar' href='item2'>
          item2
        </a>
      </header>
    );
    const skipLinks = mount(temp);
    const results = await axe(skipLinks.getDOMNode());

    // console.log('Axe violations', results.violations);

    expect(results).toHaveNoViolations();
  });
  it('Nothing is displayed when no props are passed', () => {
    const skipLinks = mount(<SkipLinks />);
    expect(skipLinks.html().trim()).toEqual('');
  });
  it('Only main nav skip links is it is only prop passed', () => {
    const skipLinks = shallow(<SkipLinks mainNavId='foo' />);
    // console.log(skipLinks.html());
    expect(skipLinks).toMatchSnapshot();
  });
  it('Only main content skip links is it is only prop passed', () => {
    const skipLinks = shallow(<SkipLinks mainContentId='bar' />);
    // console.log(skipLinks.html());
    expect(skipLinks).toMatchSnapshot();
  });
  it('Both skip links are displayed when passed', () => {
    const skipLinks = shallow(
      <SkipLinks mainNavId='foo' mainContentId='bar' />
    );
    // console.log(skipLinks.html());
    expect(skipLinks).toMatchSnapshot();
  });
});
