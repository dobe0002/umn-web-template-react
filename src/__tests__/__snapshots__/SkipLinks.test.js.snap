// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skip Links Both skip links are displayed when passed 1`] = `ShallowWrapper {}`;

exports[`Skip Links Only main content skip links is it is only prop passed 1`] = `ShallowWrapper {}`;

exports[`Skip Links Only main nav skip links is it is only prop passed 1`] = `ShallowWrapper {}`;
